src/m365/teams/commands/chat/chat-message-send.ts

```diff
@@ -0,0 +1,244 @@
+  // Therefore a retry-mechanism is implemented here. 
+  private async createConversation(memberEmails: string[], retried: number = 0): Promise<Chat> {
+    try {
+      const jsonBody = {
+        chatType: memberEmails.length > 2 ? "group" : "oneOnOne",
+        members: memberEmails.map(email => {
+          return {
+            "@odata.type": "#microsoft.graph.aadUserConversationMember",
+            roles: ["owner"],
+            "user@odata.bind": `https://graph.microsoft.com/v1.0/users/${email}`
+          };
+        })
+      };
+      
+      const requestOptions = this.getRequestOptions(`${this.resource}/v1.0/chats`, jsonBody);
+      return await request.post<Chat>(requestOptions);    
+    } 
+    catch(err: any) {
+      if (err.message?.indexOf("404") > -1 && retried < 4) {        
+        return await this.createConversation(memberEmails, retried+1);
+      }
+
+      throw err;
+    }
+  }
+
+  private async sendChatMessage(chatId: string, options: Options): Promise<void> {
+    const jsonBody = {
+      body: {
+        content: options.message
+      }
+    };
+    const requestOptions = this.getRequestOptions(`${this.resource}/v1.0/chats/${chatId}/messages`, jsonBody);
+    
+    await request.post(requestOptions);
+  }
+  
+  private async findExistingGroupChatsByMembers(expectedMemberEmails: string[]): Promise<Chat[]> {
+    const endpoint = `${this.resource}/v1.0/chats?$filter=chatType eq 'group'&$expand=members&$select=id,topic,createdDateTime,members`;
+    const foundChats: Chat[] = [];
+
+    const chats = await this.getAllChats(endpoint, []);    
+
+    for (const chat of chats) {      
+      const chatMembers = chat.members as ConversationMember[];
+      if (chatMembers.length === expectedMemberEmails.length) {
+        const chatMemberEmails = chatMembers.map(m => (m as AadUserConversationMember).email?.toLowerCase());
+
+        if (expectedMemberEmails.every(e => chatMemberEmails.some(m => m === e))) {
+          foundChats.push(chat);
+        }          
+      }
+    }
+
+    return foundChats;
+  }
+
+  private async findExistingGroupChatsByTopic(topic: string): Promise<Chat[]> {
+    const endpoint = `${this.resource}/v1.0/chats?$filter=topic eq '${encodeURIComponent(topic)}'&$expand=members&$select=id,topic,createdDateTime,chatType`;
+    const chats = await this.getAllChats(endpoint, []);
+    return chats;
+  }
+
+  private async getAllChats(url: string, items: Chat[]): Promise<Chat[]> {            
+    const requestOptions = this.getRequestOptions(url);
+    const res = await request.get<GraphResponse<Chat>>(requestOptions);
+    
+    items = items.concat(res.value);
+    
+    if (res['@odata.nextLink']) {
+      return await this.getAllChats(res['@odata.nextLink'] as string, items);              
+    }
+    else {
+      return items;
+    }            
+  }
+  
+  private getRequestOptions(url: string, data?: any): AxiosRequestConfig {
```

Comment:
For simplicity, let's avoid a single place to construct the request body and logic to conditionally add request data and build the request in place where it's needed.